include/swift/SIL/TypeSubstCloner.h

```diff
@@ -149,19 +149,9 @@ class TypeSubstCloner : public SILClonerWithScopes<ImplClass> {
     if (!Inlining) {
       FunctionRefInst *FRI = dyn_cast<FunctionRefInst>(CalleeVal);
       if (FRI && FRI->getReferencedFunction() == Inst->getFunction()) {
-        auto LoweredFnTy = Builder.getFunction().getLoweredFunctionType();
-        auto GenSig = LoweredFnTy->getGenericSignature();
-        if (GenSig) {
-          GenSig->getSubstitutions(
-              Inst->getFunction()
-                  ->getLoweredFunctionType()
-                  ->getGenericSignature()
-                  ->getSubstitutionMap(Inst->getSubstitutions()),
-              TempSubstList);
-        }
-        for (auto &Sub : TempSubstList) {
-          Sub = asImpl().getOpSubstitution(Sub);
-        }
+        for (auto Sub : Inst->getSubstitutions())
+          TempSubstList.push_back(asImpl().getOpSubstitution(Sub));
+
         SubstitutionList Subs = TempSubstList;
```

Comment:
I'm not sure about this change. The target function (i.e. the one where the cloned instructions are added) may have a different generic signature, e.g. when we perform a partial specialization. Your change assumes that the number of substitutions is always the same for the original and the specialized functions, which is wrong, IMHO.